From 2208969ea8889333d58dec5ba59920cede09f3d4 Mon Sep 17 00:00:00 2001
From: Zhao Yakui <yakui.zhao@intel.com>
Date: Thu, 26 Jul 2018 15:37:51 +0800
Subject: [PATCH 1415/1670] Kernel/drm/i915: Remove the redundant cache change
 for shared_page

The gvt-g will try to map one shared_page for PVMMIO usage so that the
guest system can pass the mmio access through PVMMIO page. Currently
it will firstly try to map the page and then change the cache type.
In fact as the shared_page is accessed through MMIO, it is meaningless
to change the cache type.
So it will remove the redundant cache change for the shared_page.

Signed-off-by: Zhao Yakui <yakui.zhao@intel.com>
Signed-off-by: Li, Fei1 <fei1.li@intel.com>
Reviewed-by: Eddie Dong <eddie.dong@intel.com>
Reviewed-by: Jiang Fei <fei.jiang@intel.com>
---
 drivers/gpu/drm/i915/gvt/acrn-gvt.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/acrn-gvt.c b/drivers/gpu/drm/i915/gvt/acrn-gvt.c
index 5c4d627..b03f326 100644
--- a/drivers/gpu/drm/i915/gvt/acrn-gvt.c
+++ b/drivers/gpu/drm/i915/gvt/acrn-gvt.c
@@ -865,16 +865,6 @@ static int acrngt_set_pvmmio(unsigned long handle, u64 start, u64 end, bool map)
 			return rc;
 		}
 
-		/* shared page is not trapped, directly pass through */
-		rc = add_memory_region(info->vm_id,
-				(pfn + mmio_size_fn) << PAGE_SHIFT,
-				shared_mfn << PAGE_SHIFT, 0x1000,
-				MEM_TYPE_WB, MEM_ACCESS_RWX);
-		if (rc) {
-			gvt_err("failed set map for gfn 0x%lx\n",
-				pfn + mmio_size_fn);
-			return rc;
-		}
 	} else {
 		mfn = acrngt_virt_to_mfn(info->vgpu->mmio.vreg);
 		rc = acrngt_map_gfn_to_mfn(handle, pfn, mfn, mmio_size_fn, map);
-- 
2.7.4

